src/components/subcomponents/upload/upload.js

```diff
@@ -1,110 +1,50 @@
-import React, { useState } from "react";
-import {
-  Grid,
-  Typography,
-  Button,
-  Paper,
-  IconButton,
-  LinearProgress
-} from "@material-ui/core";
+import React from "react";
+import FileUI from "./file-ui";
+import { Grid, Typography, Button } from "@material-ui/core";
 import { makeStyles } from "@material-ui/core/styles";
-import AttachFileIcon from "@material-ui/icons/AttachFile";
-import HighlightOffIcon from "@material-ui/icons/HighlightOff";
-import Axios from "axios";
-import UrlJoin from "url-join";
+import {
+  uploadprocess,
+  addPendingFile,
+  cancelFileUpload
+} from "../../../store/actions/submissionActions";
+import { connect } from "react-redux";
+import PublishIcon from '@material-ui/icons/Publish';
 
 const useStyles = makeStyles(theme => ({
   root: {
     marginBottom: theme.spacing(1)
   },
   input: {
     display: "none"
-  },
-  attachment: {
-    position: "relative",
-    minHeight: "10rem",
-    width: "9rem",
-    display: "flex",
-    flexFlow: "column",
-    justifyContent: "space-between",
-    alignItems: "center",
-    textAlign: "center",
-    padding: "0.5rem"
-  },
-  attachDel: {
-    position: "absolute",
-    right: "-0.5rem",
-    top: "-0.5rem"
-  },
-  attachIco: {
-    height: "5rem"
-  },
-  overflow: {
-    whiteSpace: "nowrap",
-    overflow: "hidden",
-    textOverflow: "ellipsis",
-    width: "inherit",
-    padding: "0.5rem"
-  },
-  loading: {
-    width: " 100%",
-    height: ".5rem"
   }
 }));
 
-export function UploadFiles(props) {
-  const [files, setFiles] = useState([]);
-
+export function UploadFiles({
+  files,
+  setFiles,
+  upload,
+  uploadFiles,
+  addFile,
+  cancelUpload
+}) {
   const onAddFile = e => {
-    const toUpload = [];
-    Array.from(e.target.files).forEach(file => {
-      toUpload.push({ data: file, upload: "PENDING" });
+    const toUpload = {};
+    let fileID;
+    Array.from(e.target.files).map(file => {
+      fileID = Math.floor(Math.random() * 99 + 1);
+      fileID = "" + fileID + file.size;
+      toUpload[fileID] = file;
+      addFile(fileID);
     });
-    setFiles([...files, ...toUpload]);
-    e.target.files = null
+    setFiles({ ...files, ...toUpload });
+    e.target.files = null;
   };
-  const onDelete = (file) => {
-    let delete_Array = files
-    delete_Array.splice(files.indexOf(file),1)
-    setFiles([...delete_Array])
+  const onDelete = fileID => {
+    let delete_obj = files;
+    delete delete_obj[fileID];
```

Comment:
uh oh, you're mutating your props, if you want mutate the array, make a copy of it 

```
const delete_obj = [...files]
delete delete_obj[fileID];
```

However, I'm wondering why you need this prop at all, doesn't redux hold all this information for you in `uploadFiles`?

Reply:
Yep, had a mix of things. I've cleaned it up now